src/Commands.pm

```diff
@@ -6564,113 +6546,144 @@ sub cmdMail {
+		unless ($args[1] =~ /^\d+$/) {
+			error T("Syntax Error in function 'mail read' (Mailbox)\n" .
+				"Usage: mail read <mail #>\n");
+		} elsif (!defined $mailList) {
+			warning T("Your Mailbox is not open. Use the command 'mail open'.\n");
+		} elsif (!$mailList->[$args[1]]) {
+				warning TF("No mail found with index: %s\n", $args[1]);
+		} elsif ($mailList->[$args[1]]->{mailID}) {
+			$messageSender->sendMailRead($mailList->[$args[1]]->{mailID});
+		}
+
+	} elsif ($args[0] eq 'get') {
+		unless ($args[1] =~ /^\d+$/) {
+			error T("Syntax Error in function 'mail get' (Mailbox)\n" .
+				"Usage: mail get <mail #>\n");
+		} elsif (!defined $mailList) {
+			warning T("Your Mailbox is not open. Use the command 'mail open'.\n");
+		} elsif (!$mailList->[$args[1]]) {
+				warning TF("No mail found with index: %s\n", $args[1]);
+		} elsif ($mailList->[$args[1]]->{mailID}) {
+			$messageSender->sendMailGetAttach($mailList->[$args[1]]->{mailID});
+		}
+
+	} elsif ($args[0] eq 'setzeny') {
+		if ($args[1] =~ /^\d+$/) {
+			$messageSender->sendMailSetAttach($args[1], undef);
+		} elsif ($args[1] eq 'none') {
+			$messageSender->sendMailOperateWindow(2);
 		} else {
-			error T("Syntax error in function 'mw' (mailbox window)\n" .
-			"Usage: mw [0|1|2] (0:write, 1:take item back, 2:zeny input ok)\n");
+			error T("Syntax Error in function 'mail setzeny' (Mailbox)\n" .
+				"Usage: mail setzeny <amount|none>\n");
 		}
 
-	# mail attachment control
-	} elsif ($cmd eq 'ma') {
-		if ($args[0] eq "get" && $args[1] =~ /^\d+$/) {
-			unless ($mailList->[$args[1]]->{mailID}) {
-				if (@{$mailList}) {
-					message TF("No mail found with index: %s. (might need to re-open mailbox)\n", $args[1]), "info";
-				} else {
-					message T("Mailbox has not been opened or is empty.\n"), "info";
-				}
-			} else {
-				$messageSender->sendMailGetAttach($mailList->[$args[1]]->{mailID});
-			}
-		} elsif ($args[0] eq "add") {
-			unless ($args[2] =~ /^\d+$/) {
-				message T("Usage: ma add [zeny <amount>]|[item <amount> (<item #>|<item name>)]\n"), "info";
-			} elsif ($args[1] eq "zeny") {
-				$messageSender->sendMailSetAttach($args[2], undef);
-			} elsif ($args[1] eq "item" && defined $args[3]) {
-				my $item = Actor::Item::get($args[3]);
-				if ($item) {
-					my $serverIndex = $item->{ID};
-					$messageSender->sendMailSetAttach($args[2], $serverIndex);
-				} else {
-					message TF("Item with index or name: %s does not exist in inventory.\n", $args[3]), "info";
-				}
+	} elsif ($args[0] eq 'add') {
+		unless (defined $args[1]) {
+			error T("Syntax Error in function 'mail add' (Mailbox)\n" .
+				"Usage: mail add <item #> <amount>\n");
+		} elsif ($args[1] eq 'none') {
+			$messageSender->sendMailOperateWindow(1);
+		} else {
+			my $item = Actor::Item::get($args[1]);
+			if ($item) {
+				my $amount = $args[2] ? $args[2] : $item->{amount};
+				warning TF("Attention: Inventory Item '%s' is equipped.\n", $item->{name}) if ($item->{equipped});
```

Comment:
should we add a return after this? do the server accept equipped items?

Reply:
`return` is not needed
```
i eq
------------------- Inventory --------------------
-- Equipment (Equipped) --
2    Cotton Shirt -- Armor (16)
--------------------------------------------------

mail add 2
Attention: Inventory Item 'Cotton Shirt' is equipped.
Succeeded to attach item: Cotton Shirt (2).
Inventory Item Removed: Cotton Shirt (2) x 1

i eq
------------------- Inventory --------------------
-- Equipment (Equipped) --
--------------------------------------------------
```